Assets/Scripts/Enemies/Pufferbomb.cs

```diff
@@ -0,0 +1,52 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class Pufferbomb : MonoBehaviour
+{
+    public int damage; //damage dealt to Player
+    public float distance; //distance between Pufferbomb and Player for Pufferbomb to explode
+
+    private ParticleSystem explosion;
+    private MeshRenderer mesh;
+    private bool isExploding; //check that Pufferbomb is already exploding
+    private bool isShot; //check if Player already shot Pufferbomb
+
+    void Start()
+    {
+        explosion = transform.GetComponent<ParticleSystem>();
+        mesh = transform.GetComponent<MeshRenderer>();
+
+        isExploding = false;
+        isShot = false;
+    }
+
+    void Update()
+    {
+        if (!isExploding && !isShot && Vector3.Distance(PlayerInfo.singleton.transform.position, transform.position) <= distance)
```

Comment:
It's not very good to calculate the distance within the update function. I would add a new sphere collider on the pufferbomb prefab that's a trigger and then you can use OnTriggerEnter() to determine when the player is close enough for it to explode. Also i'm unsure why but when testing the puffer fish was not exploding when I ran into it

Reply:
I added the sphere collider trigger. I'm not sure if my implementation is efficient though, so would like to know if it needs to be revised.